File: Array/Arithmetic Slices/code.py
# https://leetcode.com/problems/arithmetic-slices/
class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        if len(A)<3: return 0
        diff=None
        ret=0
        count=1
        for i in range(1,len(A)):
            if diff!=A[i]-A[i-1]: 
                count=2
                diff=A[i]-A[i-1]
            else:
                count+=1
                if count >=3: ret+=(count-3)+1
            # print(count,ret)
        return ret

File: String/Longest Word in Dictionary through Deleting/code.java
# https://leetcode.com/problems/longest-word-in-dictionary-through-deleting/
class Solution {
    public String findLongestWord(String s, List<String> d) {
        String ret="";
        int max_val=0;
        int l=s.length();
        for(int i=0;i<d.size(); i++){
            int start=0;
            int at_l = d.get(i).length();
            if (at_l <max_val) continue;
            int check = 0;
            for (int j=0; j<l && start < at_l && l-j>=at_l-start; j++){
                if ((s.charAt(j))==(d.get(i).charAt(start))){
                    if (check==0 && start<max_val){
                        check = (int)(d.get(i).charAt(start))-(int)(ret.charAt(start));
                    } 
                    start+=1;
                }
            }
            if (start == at_l && (max_val<at_l || (max_val==at_l && check<0))) {
                ret=d.get(i);
                max_val=at_l;
            }
        }
        return ret;
    }
}